packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,220 @@
+
+                return false;
+            }
+
+            const specifiedType = iconType.split('/')[1];
+
+            /** Handling for the corrupt rawContent */
+            if (rawContent !== null) {
+                const image: ImageTypeResult | null = imageType(rawContent);
+
+                if (image !== null) {
+                    const isValidType: boolean = allowedTypes.some((item) => {
+                        return item === image.ext;
+                    });
+
+                    if (specifiedType !== image.ext) {
+                        const message = `Real image type (${image.ext}) do not match with specified type (${specifiedType})`;
+
+                        context.report(iconPath, message);
+                    }
+
+                    if (isValidType) {
+                        return true;
+                    }
+                }
+
+                const message = `Icon should be a valid image type ${JSON.stringify(allowedTypes)}`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            return false;
+        };
+
+        /**
+         *
+         * @param iconSizes Sizes specified in the manifest file
+         * @param iconRawData
+         * @param iconPath full path to the icon file
+         */
+        const validateSizes = (iconSizes: string | undefined, iconRawData: Buffer, iconPath: string) => {
+            if (iconSizes === undefined) {
+                context.report(iconPath, `Sizes not specifed for icon`);
+
+                return false;
+            }
+            /**
+             * sizes can be the string 'any' OR
+             * two non-negative integers without leading 0s and separated by 'x' like 144x144
+             */
+            if (iconSizes === 'any') {
+                return false;
+            }
+            const specifiedSize = iconSizes.split('x');
+
+            const realImage = getImageData(iconRawData);
+
+            const realSize = [realImage.width.toString(), realImage.height.toString()];
+
+            const sizesMatch = JSON.stringify(specifiedSize) === JSON.stringify(realSize);
+
+            if (!sizesMatch) {
+                const message = `Real image size (${JSON.stringify(realSize)}) do not match with specified size (${specifiedSize})`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            return true;
+        };
+
+        const hasRequiredSizes = (validSizes: string[], icons: string) => {
+            const requiredSizes = ['192x192', '512x512'];
+
+            const requiredSizesNotFound = requiredSizes.filter((size) => {
+                return !validSizes.includes(size);
+            });
+
+            if (requiredSizesNotFound.length > 0) {
+                const message = `Required sizes ${JSON.stringify(requiredSizesNotFound)} not found.`;
+
+                context.report(icons, message);
```

Comment:
Each report that can reference a particular point in the manifest should include a location (see `hint-manifest-app-name` for an example of using the `getLocation` helper on the `ManifestParsed` event to do this). Also update the tests to validate this location. 
```suggestion
                context.report(icons, message, { location });
```

Reply:
`ProblemLocation`s added to the reports.

Reply:
@shivangg Thanks! But it looks like you may still need to push those changes - I don't see them.

Reply:
Yes, pushing them. I thought pushing before completing the full set of proposed changes will unnecessarily startup the CI pipeline. And not resolving the comments makes it difficult to make sense of what changes still need to be handled by me.

Am I missing some best practice?

Reply:
Don't worry about CI, push when you have changes ready so conversation status are in sync with the code and we know when to review. If needed I can cancel scheduled builds.